src/lib.rs

```diff
@@ -233,6 +219,20 @@ impl Keccak {
         }
     }
 
+    fn a_bytes(&self) -> &[u8] {
```

Comment:
I wonder if the compiler could reason better about bounds checks if we returned an `&[u8; PLEN * 8]` here (and likewise in `a_mut_bytes`).

Then the body of each function is just a transmute and we save a pointer's worth of stack space :)